### PR TITLE
Add prefix to cli license scanned paths

### PR DESCRIPTION
**Pull request by spatten**
# Overview

[closes ANE-215](https://fossa.atlassian.net/browse/ANE-215)

We want to show the paths to the files being scanned in exactly the same way for archive uploads and cli-license scans.

To do this for cli-license scans, we will add a `--path-prefix` argument when we shell out to `themis-cli`.

This PR figures out what that path prefix should be and adds it to the invocation.

## Acceptance criteria

* Paths for cli-side license scanned archived vendored dependencies should be prefixed correctly
* Paths for cli-side license scanned unarchived vendored dependencies should be prefixed correctly

## Testing plan

Run a cli-side license scan on an archive and unarchived vendored dependency. This testing plan uses example projects from current master on `https://github.com/fossas/example-projects`.

You'll need to update to a version of `themis-cli` that has the `--path-prefix` flag:

```
./vendor_download.sh
```

First, let's scan `example-projects/fossa-deps/vendored-dependencies`.

That project contains a tar file called `hello-1.0.0.2.tar.gz` in the vendored directory. If you unarchive that tar file, it looks like this:

```
├── hello-1.0.0.2
│   ├── LICENSE
│   ├── Setup.hs
│   ├── hello.cabal
│   └── src
│       └── hello.hs
```

So what we hope to see if the path to the LICENSE file looking like `vendored/hello-1.0.0.2/LICENSE`.

Let's do the scan:

```
FOSSA_API_KEY=$FOSSA_LOCAL_API_KEY cabal run fossa -- analyze -e http://localhost:9578 --experimental-native-license-scan ~/fossa/example-projects/fossa-deps/vendored-dependencies 
```

Once the scan is done and the dependency is built, go to the licenses tab and click on the BSD-2-clause license.

On current master, you see this:

Screenshot 2022-04-19 at 2 14 46 PM

With this branch, you see this:

Screenshot 2022-04-19 at 2 13 46 PM

Do the same thing with an unarchived dependency, found in `fossa-deps/unarchived-vendored-dependencies`.

```
# tree fossa-deps/unarchived-vendored-dependencies/vendor
fossa-deps/unarchived-vendored-dependencies/vendor
└── hello-1.0.0.2
    ├── LICENSE
    ├── Setup.hs
    ├── hello.cabal
    └── src
        └── hello.hs
```

```
FOSSA_API_KEY=$FOSSA_LOCAL_API_KEY cabal run fossa -- analyze -e http://localhost:9578 --experimental-native-license-scan ~/fossa/example-projects/fossa-deps/unarchived-vendored-dependencies                        
```

On master, you can see it's just showing the path as `LICENSE`:

Screenshot 2022-04-19 at 2 20 04 PM

On this branch it shows the path to the LICENSE file properly:

Screenshot 2022-04-19 at 2 24 37 PM

## References

[closes ANE-215](https://fossa.atlassian.net/browse/ANE-215)

## Checklist

- [ ] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [ ] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [ ] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [ ] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

### CODE REVIEW

File: src/App/Fossa/LicenseScanner.hs
Comment (meghfossa) [inline src/App/Fossa/LicenseScanner.hs:153]: [no action] you can also do `mempty` or `Data.Text.empty` instead of `""`

**Review by meghfossa — APPROVED** [review]
Let's do a release with this change (strong preference). 

I did not test the code against core, but the changes look simple enough, and the test plan (with screenshots) looks great.

**Comment by spatten** [comment]
> Let's do a release with this change (strong preference).

Would you be okay with making a release after #899 is merged? 

There are no users currently using this functionality, so I think that there's no harm with waiting until both are merged

**Comment by meghfossa** [comment]
> > Let's do a release with this change (strong preference).
> 
> Would you be okay with making a release after #899 is merged?
> 
> There are no users currently using this functionality, so I think that there's no harm with waiting until both are merged

sounds good 👍

**Pull request merged**
